bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/LibertyDevPlugin.java

```diff
@@ -55,17 +85,19 @@ public void start(BundleContext context) throws Exception {
         props.put(DebugOptions.LISTENER_SYMBOLICNAME, LibertyDevPlugin.DEBUG_OPTIONS_ID);
         context.registerService(DebugOptionsListener.class.getName(), new Trace(), props);
 
-        DevModeOperations.getInstance().getProjectModel().createNewCompleteWorkspaceModelWithClassify();
+        // Register a workspace listener for cleanup.
+        registerListeners();
 
-        resourceChangeListener = new LibertyResourceChangeListener();
-        ResourcesPlugin.getWorkspace().addResourceChangeListener(resourceChangeListener, IResourceChangeEvent.PRE_BUILD);
+        // Classify all projects in the workspace.
+        DevModeOperations.getInstance().getProjectModel().createNewCompleteWorkspaceModelWithClassify();
```

Comment:
This should not really be an issue because this all happens during plugin startup, but in case i am missing something, i  moved it back. I also fixes the other two issues (printStackTrace left in the code, and the typo).